File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.java
class Solution {
    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int ans=0;
        int left=0;
        int n=arr.length;
        int cur_sum=0;
        for(int right=0;right<n;right++){
            cur_sum+=arr[right];
            if (right-left+1>k){
                cur_sum-=arr[left];
                left++;
            }
            if (right-left+1==k){
                if (cur_sum>=threshold*k){
                    ans++;

                }
            }
        }
        return ans;
        
    }
}

File: 1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold/1343-number-of-sub-arrays-of-size-k-and-average-greater-than-or-equal-to-threshold.py
class Solution:
    def numOfSubarrays(self, nums: List[int], k: int, threshold: int) -> int:
        ans=0
        left=0
        summ=0
        for right in range(len(nums)):
            summ+=nums[right]
            if right-left+1>k:
                summ-=nums[left]
                left+=1

            if right-left+1==k:
                if summ//k>=threshold:
                    ans+=1    
        return ans            
        